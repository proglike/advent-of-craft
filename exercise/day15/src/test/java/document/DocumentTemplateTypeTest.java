package document;

import io.vavr.collection.List;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class DocumentTemplateTypeTest {

    @Test
    void approvalTest() {

        List<String> documentTypes = List.of(
                "DEER",
                "AUTP",
                "AUTM",
                "SPEC",
                "GLPP",
                "GLPM",
                "Autre Document Type");

        List<String> recordTypes = List.of((RecordType.values()))
                .map(RecordType::name)
                .append("Autre Record Type");

        CombinationApprovals.verifyAllCombinations(
                DocumentTemplateType::fromDocumentTypeAndRecordType,
                documentTypes.toJavaArray(String[]::new),
                recordTypes.toJavaArray(String[]::new));
    }

}