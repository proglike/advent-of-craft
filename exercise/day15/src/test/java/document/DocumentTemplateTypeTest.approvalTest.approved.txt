[DEER, INDIVIDUAL_PROSPECT] => DEERPP 
[DEER, LEGAL_PROSPECT] => DEERPM 
[DEER, ALL] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[DEER, Autre Record Type] => java.lang.IllegalArgumentException: No enum constant document.RecordType.Autre Record Type 
[AUTP, INDIVIDUAL_PROSPECT] => AUTP 
[AUTP, LEGAL_PROSPECT] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[AUTP, ALL] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[AUTP, Autre Record Type] => java.lang.IllegalArgumentException: No enum constant document.RecordType.Autre Record Type 
[AUTM, INDIVIDUAL_PROSPECT] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[AUTM, LEGAL_PROSPECT] => AUTM 
[AUTM, ALL] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[AUTM, Autre Record Type] => java.lang.IllegalArgumentException: No enum constant document.RecordType.Autre Record Type 
[SPEC, INDIVIDUAL_PROSPECT] => SPEC 
[SPEC, LEGAL_PROSPECT] => SPEC 
[SPEC, ALL] => SPEC 
[SPEC, Autre Record Type] => java.lang.IllegalArgumentException: No enum constant document.RecordType.Autre Record Type 
[GLPP, INDIVIDUAL_PROSPECT] => GLPP 
[GLPP, LEGAL_PROSPECT] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[GLPP, ALL] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[GLPP, Autre Record Type] => java.lang.IllegalArgumentException: No enum constant document.RecordType.Autre Record Type 
[GLPM, INDIVIDUAL_PROSPECT] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[GLPM, LEGAL_PROSPECT] => GLPM 
[GLPM, ALL] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[GLPM, Autre Record Type] => java.lang.IllegalArgumentException: No enum constant document.RecordType.Autre Record Type 
[Autre Document Type, INDIVIDUAL_PROSPECT] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[Autre Document Type, LEGAL_PROSPECT] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[Autre Document Type, ALL] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
[Autre Document Type, Autre Record Type] => java.lang.IllegalArgumentException: Invalid Document template type or record type 
